backends/tc/backend.cpp

```diff
@@ -354,6 +354,32 @@ void ConvertToBackendIR::updateDefaultMissAction(const IR::P4Table *t, IR::TCTab
                 if (defaultActionProperty->isConstant) {
                     tabledef->setDefaultMissConst(true);
                 }
+                bool directionParamPresent = false;
+                auto paramList = actionCall->action->getParameters();
+                for (auto param : paramList->parameters) {
+                    if (param->direction != IR::Direction::None) directionParamPresent = true;
+                }
+                if (!directionParamPresent) {
+                    auto i = 0;
+                    for (auto param : paramList->parameters) {
+                        auto defaultParam = new IR::TCDefaultActionParam();
+                        defaultParam->setParamName(param->name.originalName);
+                        auto defaultArg = methodexp->arguments->at(i++);
+                        if (defaultArg->expression->is<IR::Constant>()) {
```

Comment:
Why not use if (auto constVal =  defaultArg->expression->to<IR::Constant>())

Reply:
Updated the code